docs/source/machine_translation.rst

```diff
@@ -0,0 +1,255 @@
+file is generated on both source and target dataset.
+You can generate it by running following script::
+
+  neuralmonkey/lib/subword_nmt/learn_bpe.py -s 50000 < DATA > merge_file.bpe
+
+With the data from this tutorial it would be the following command::
+
+  paste Batch1a_en.txt Batch1a_cs.txt |
+  neuralmonkey/lib/subword_nmt/learn_bpe.py -s 8000 >
+  exp-nm-mt/data/merge_file.bpe
+
+You can change number of merges, this number is equivalent to the size of the
+vocabulary. Do not forget that as an input is the file containing both source
+and target sides.
+
+
+
+
+Part II. - The Model Configuration
+----------------------------------
+
+In this section, we will go through setting the configuration file
+``translation.ini`` needed for
+the training of the machine translation task. We will mention only
+differences from the main Post_editing tutorial
+
+1 - Datasets
+************
+
+For training, we prepare two datasets. Since we will be using BPE, we need to
+ define the preprocessor. The configuration of the datasets looks like this::
+
+  [train_data]
+  class=config.utils.dataset_from_files
+  s_source=exp-nm-mt/data/train/Batch1a_en.txt.gz
+  s_target=exp-nm-mt/data/train/Batch1a_cs.txt.gz
+  preprocessor=<bpe_preprocess>
+
+  [val_data]
+  class=config.utils.dataset_from_files
+  s_source=exp-nm-mt/data/dev/Batch2a_en.txt.gz
+  s_target=exp-nm-mt/data/dev/Batch2a_cs.txt.gz
+  preprocessor=<bpe_preprocess>
+
+2 - Preprocessor and postprocessor
+**********************************
+
+Wee need to tell the Neural Monkey how it should handle preprocessing and
+postprocessing due to the BPR::
+
+  [bpe_preprocess]
+  class=processors.bpe.BPEPreprocessor
+  merge_file=exp-nm-mt/data/merge_file.bpe
+
+  [bpe_postprocess]
+  class=processors.bpe.BPEPostprocessor
+
+
+3 - Vocabularies
+****************
+
+For both encoder and decoder we will use shared vocabulary created from BPE
+merges::
+
+  [shared_vocabulary]
+  class=vocabulary.from_bpe
+  path=exp-nm-mt/data/merge_file.bpe
+
+4 - Encoder and Decoder
+************
+
+The encoder and decored is similar to Post-Editing ones::
+
+  [encoder]
+  class=encoders.sentence_encoder.SentenceEncoder
+  name=sentence_encoder
+  rnn_size=300
+  max_input_len=50
+  embedding_size=300
+  dropout_keep_prob=0.8
+  attention_type=decoding_function.Attention
+  data_id=source
+  vocabulary=<shared_vocabulary>
+
+  [decoder]
+  class=decoders.decoder.Decoder
+  name=decoder
+  encoders=[<encoder>]
+  rnn_size=256
+  embedding_size=300
+  use_attention=True
+  dropout_keep_prob=0.8
+  data_id=target
+  vocabulary=<shared_vocabulary>
+  max_output_len=50
+
+
+5 - Training sections
+**********************
+
+Following sections are described more in detail in Post-editing task::
+
+  [trainer]
+  class=trainers.cross_entropy_trainer.CrossEntropyTrainer
+  decoders=[<decoder>]
+  l2_weight=1.0e-8
+
+  [runner]
+  class=runners.runner.GreedyRunner
+  decoder=<decoder>
+  output_series=series_named_greedy
```

Comment:
Runner musí mít post-processor. (Čili, tak jak jsi měl konfigurák, tak se ti to zdálo v pohodě, protože BPE se vůbec nepoužilo, ani na vstupu, ani na výstupu.)